SPEC.md

```diff
@@ -0,0 +1,1592 @@
+with data errors corrected, and types mapped to externally defined schema concepts:
+
+```
+select
+    ga4gh_type(id, '$ref:https://schemablocks.org/schemas/sb-phenopackets/current/Person.json#properties/individualId') as id,
+    ga4gh_type(
+        case when blood_type = '' then null
+        else cast(row(
+            case regexp_extract(blood_type, '(\w+)([+-])', 1)
+                when '0' then 'HP:0032442' -- source data has '0' where it should have 'O'
+                when 'O' then 'HP:0032442'
+                when 'A' then 'HP:0032370'
+                when 'B' then 'HP:0032440'
+                when 'AB' then 'HP:0032441'
+                else 'error'
+            end,
+            case regexp_extract(blood_type, '(\w+)([+-])', 1)
+                when '0' then 'O'
+                else regexp_extract(blood_type, '(\w+)([+-])', 1)
+            end
+        )
+        as row(id varchar, label varchar))
+    end, 
+    '$ref:https://schemablocks.org/schemas/playground/current/BloodGroup.json') as blood_group
+from pgpc.public.participant
+```
+
+The Data Connect service responds with the following table:
+
+```
+{
+  "data_model": {
+    "description": "Schema specified by query",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "properties": {
+      "id": {
+        "$ref": "https://schemablocks.org/schemas/sb-phenopackets/current/Person.json#properties/individualId"
+      },
+      "blood_group": {
+        "$ref": "https://schemablocks.org/schemas/playground/current/BloodGroup.json"
+      }
+    }
+  },
+  "data": [
+    {
+      "id": "PGPC-44",
+      "blood_group": {"id": "HP:0032442", "label": "O"}
+    },
+    {
+      "id": "PGPC-46",
+      "blood_group": {"id": "HP:0032441", "label": "AB"}
+    }
+  ]
+}
+```
+
+## SQL Functions
+
+Data Connect specification is implementation-agnostic and does not prescribe use of a relational database or a particular database technology. However, for convenience, its SQL dialect has been selected for compatibility with current major open source database platforms including Trino, PostgreSQL, MySQL, and well as BigQuery. There are occasional name or signature differences, but a Data Connect API implementation atop any of the major database platforms should be able to pass through queries that use the functions listed below with only minor tweaks.
+
+With Trino (formerly PrestoSQL) being a popular database choice in Data Connect implementations, we've chosen its grammar as the basis for the grammar supported by Data Connect. Functions below are a subset of those available in Trino 341, and must behave according to the Trino documentation in a conforming Data Connect implementation. To assist with implementations directly on other database platforms, the [Trino Functions Support Matrix](https://docs.google.com/document/d/1y51qNuoe2ELX9kCOyQbFB4jihiKt2N8Qcd6-zzadIvk) captures the differences between platforms in granular detail. 
+
+*   **Logical Operators**
```

Comment:
This entire section seems to have broken formatting, we should fix this. Not sure what the significance of `*` at the end of some lines is, some identifiers are not escaped or backticked correctly, the final section is off and JSON functions has broken formatting.

Reply:
Looks automated. @jfuerth Do you remember where we transferred this from?